### DISCUSSION TITLE
bind update 9.16.10 (#6309)

### DISCUSSION THREAD

**Discussion by AndWac** [OpenIndiana/oi-userland#6311] [Q&A]
Hi guys,

I'd like somebody to check my PR #6309 before I merge it.
I cannot carefully test it myself so I am looking for testers. Maybe somebody can merge it into his/her own branch, create and install a package and check whether the new version really works for him/her.

Regards,
Andreas

**Comment by AndWac** [comment]
If nobody cares enough to check it I will merge the PR soon.

**Reply by balrog-nona** [reply]
Hello Andreas, I haven't noticed your post here. I'll try it and let you know what it looks like on my machine.

**Comment by balrog-nona** [comment]
So far, I'm not able to perform `gmake test`.
First time, it went like this:
```
/bin/rm -f -rf /export/home/nona/oi-userland/components/network/bind/build/test/amd64
/bin/mkdir -p /export/home/nona/oi-userland/components/network/bind/build/test/amd64
cd /export/home/nona/oi-userland/components/network/bind/build/amd64/bin/tests/system ; pfexec sh ./ifconfig.sh up ;
ifconfig: cannot plumb lo0:1: Insufficient user authorizations
ifconfig: error: lo0:1: no such interface
ifconfig: cannot plumb lo0:1: Insufficient user authorizations
ifconfig: error: lo0:1: no such interface
ifconfig: cannot plumb lo0:2: Insufficient user authorizations
ifconfig: error: lo0:2: no such interface
ifconfig: cannot plumb lo0:2: Insufficient user authorizations
ifconfig: error: lo0:2: no such interface
ifconfig: cannot plumb lo0:3: Insufficient user authorizations
ifconfig: error: lo0:3: no such interface
ifconfig: cannot plumb lo0:3: Insufficient user authorizations
ifconfig: error: lo0:3: no such interface
ifconfig: cannot plumb lo0:4: Insufficient user authorizations
ifconfig: error: lo0:4: no such interface
ifconfig: cannot plumb lo0:4: Insufficient user authorizations
ifconfig: error: lo0:4: no such interface
ifconfig: cannot plumb lo0:5: Insufficient user authorizations
ifconfig: error: lo0:5: no such interface
ifconfig: cannot plumb lo0:5: Insufficient user authorizations
ifconfig: error: lo0:5: no such interface
ifconfig: cannot plumb lo0:6: Insufficient user authorizations
ifconfig: error: lo0:6: no such interface
ifconfig: cannot plumb lo0:6: Insufficient user authorizations
ifconfig: error: lo0:6: no such interface
ifconfig: cannot plumb lo0:7: Insufficient user authorizations
ifconfig: error: lo0:7: no such interface
ifconfig: cannot plumb lo0:7: Insufficient user authorizations
ifconfig: error: lo0:7: no such interface
ifconfig: cannot plumb lo0:8: Insufficient user authorizations
ifconfig: error: lo0:8: no such interface
ifconfig: cannot plumb lo0:8: Insufficient user authorizations
ifconfig: error: lo0:8: no such interface
ifconfig: cannot plumb lo0:9: Insufficient user authorizations
ifconfig: error: lo0:9: no such interface
ifconfig: cannot plumb lo0:9: Insufficient user authorizations
ifconfig: error: lo0:9: no such interface
ifconfig: cannot plumb lo0:10: Insufficient user authorizations
ifconfig: error: lo0:10: no such interface
ifconfig: cannot plumb lo0:10: Insufficient user authorizations
ifconfig: error: lo0:10: no such interface
ifconfig: cannot plumb lo0:11: Insufficient user authorizations
ifconfig: error: lo0:11: no such interface
ifconfig: cannot plumb lo0:11: Insufficient user authorizations
ifconfig: error: lo0:11: no such interface
ifconfig: cannot plumb lo0:12: Insufficient user authorizations
ifconfig: error: lo0:12: no such interface
ifconfig: cannot plumb lo0:12: Insufficient user authorizations
ifconfig: error: lo0:12: no such interface
ifconfig: cannot plumb lo0:21: Insufficient user authorizations
ifconfig: error: lo0:21: no such interface
ifconfig: cannot plumb lo0:21: Insufficient user authorizations
ifconfig: error: lo0:21: no such interface
ifconfig: cannot plumb lo0:22: Insufficient user authorizations
ifconfig: error: lo0:22: no such interface
ifconfig: cannot plumb lo0:22: Insufficient user authorizations
ifconfig: error: lo0:22: no such interface
```
and I got this result:
```
Test results in /export/home/nona/oi-userland/components/network/bind/build/test/amd64/test-64-results
Differences found.
--- /export/home/nona/oi-userland/components/network/bind/test/results-64.master        2021-01-05 13:02:39.128862922 +0000
+++ /export/home/nona/oi-userland/components/network/bind/build/test/amd64/results-64.snapshot  2021-01-05 13:09:39.372941869 +0000
@@ -1,9 +0,0 @@
-R:cds:FAIL
-R:nsupdate:FAIL
-R:tsig:FAIL
-R:tsiggss:FAIL
-I:System test result summary:
-I:      4 FAIL
-I:     74 PASS
-I:      3 SKIPPED
-I:     19 UNTESTED
gmake: *** [/export/home/nona/oi-userland/make-rules/configure.mk:222: /export/home/nona/oi-userland/components/network/bind/build/amd64/.tested-and-compared] Error 2
```
Second time, I ran `gmake test` as sudo, and I got the same result although without the `Insufficient user authorizations` and `no such interface` messages.
In both cases, `build/test/amd64/results-64.snapshot` was an empty file.
Can you tell me how to run tests properly?

**Comment by AndWac** [comment]
Am 05.01.21 um 15:13 schrieb Nona Hansel:
>
> So far, I'm not able to perform |gmake test|.
> First time, it went like this:
>
> |/bin/rm -f -rf
> /export/home/nona/oi-userland/components/network/bind/build/test/amd64
> /bin/mkdir -p
> /export/home/nona/oi-userland/components/network/bind/build/test/amd64
> cd
> /export/home/nona/oi-userland/components/network/bind/build/amd64/bin/tests/system
> ; pfexec sh ./ifconfig.sh up ; ifconfig: cannot plumb lo0:1:
> Insufficient user authorizations ifconfig: error: lo0:1: no such
> interface ifconfig: cannot plumb lo0:1: Insufficient user
> authorizations ifconfig: error: lo0:1: no such interface ifconfig:
> cannot plumb lo0:2: Insufficient user authorizations ifconfig: error:
> lo0:2: no such interface ifconfig: cannot plumb lo0:2: Insufficient
> user authorizations ifconfig: error: lo0:2: no such interface
> ifconfig: cannot plumb lo0:3: Insufficient user authorizations
> ifconfig: error: lo0:3: no such interface ifconfig: cannot plumb
> lo0:3: Insufficient user authorizations ifconfig: error: lo0:3: no
> such interface ifconfig: cannot plumb lo0:4: Insufficient user
> authorizations ifconfig: error: lo0:4: no such interface ifconfig:
> cannot plumb lo0:4: Insufficient user authorizations ifconfig: error:
> lo0:4: no such interface ifconfig: cannot plumb lo0:5: Insufficient
> user authorizations ifconfig: error: lo0:5: no such interface
> ifconfig: cannot plumb lo0:5: Insufficient user authorizations
> ifconfig: error: lo0:5: no such interface ifconfig: cannot plumb
> lo0:6: Insufficient user authorizations ifconfig: error: lo0:6: no
> such interface ifconfig: cannot plumb lo0:6: Insufficient user
> authorizations ifconfig: error: lo0:6: no such interface ifconfig:
> cannot plumb lo0:7: Insufficient user authorizations ifconfig: error:
> lo0:7: no such interface ifconfig: cannot plumb lo0:7: Insufficient
> user authorizations ifconfig: error: lo0:7: no such interface
> ifconfig: cannot plumb lo0:8: Insufficient user authorizations
> ifconfig: error: lo0:8: no such interface ifconfig: cannot plumb
> lo0:8: Insufficient user authorizations ifconfig: error: lo0:8: no
> such interface ifconfig: cannot plumb lo0:9: Insufficient user
> authorizations ifconfig: error: lo0:9: no such interface ifconfig:
> cannot plumb lo0:9: Insufficient user authorizations ifconfig: error:
> lo0:9: no such interface ifconfig: cannot plumb lo0:10: Insufficient
> user authorizations ifconfig: error: lo0:10: no such interface
> ifconfig: cannot plumb lo0:10: Insufficient user authorizations
> ifconfig: error: lo0:10: no such interface ifconfig: cannot plumb
> lo0:11: Insufficient user authorizations ifconfig: error: lo0:11: no
> such interface ifconfig: cannot plumb lo0:11: Insufficient user
> authorizations ifconfig: error: lo0:11: no such interface ifconfig:
> cannot plumb lo0:12: Insufficient user authorizations ifconfig: error:
> lo0:12: no such interface ifconfig: cannot plumb lo0:12: Insufficient
> user authorizations ifconfig: error: lo0:12: no such interface
> ifconfig: cannot plumb lo0:21: Insufficient user authorizations
> ifconfig: error: lo0:21: no such interface ifconfig: cannot plumb
> lo0:21: Insufficient user authorizations ifconfig: error: lo0:21: no
> such interface ifconfig: cannot plumb lo0:22: Insufficient user
> authorizations ifconfig: error: lo0:22: no such interface ifconfig:
> cannot plumb lo0:22: Insufficient user authorizations ifconfig: error:
> lo0:22: no such interface |
>
> and I got this result:
>
> |Test results in
> /export/home/nona/oi-userland/components/network/bind/build/test/amd64/test-64-results
> Differences found. ---
> /export/home/nona/oi-userland/components/network/bind/test/results-64.master
> 2021-01-05 13:02:39.128862922 +0000 +++
> /export/home/nona/oi-userland/components/network/bind/build/test/amd64/results-64.snapshot
> 2021-01-05 13:09:39.372941869 +0000 @@ -1,9 +0,0 @@ -R:cds:FAIL
> -R:nsupdate:FAIL -R:tsig:FAIL -R:tsiggss:FAIL -I:System test result
> summary: -I: 4 FAIL -I: 74 PASS -I: 3 SKIPPED -I: 19 UNTESTED gmake:
> *** [/export/home/nona/oi-userland/make-rules/configure.mk:222:
> /export/home/nona/oi-userland/components/network/bind/build/amd64/.tested-and-compared]
> Error 2 |
>
> Second time, I ran |gmake test| as sudo, and I got the same result
> although without the |Insufficient user authorizations| and |no such
> interface| messages.
> In both cases, |build/test/amd64/results-64.snapshot| was an empty file.
> Can you tell me how to run tests properly?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/discussions/6311#discussioncomment-262571>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADET4N5DAIJAT6OC6NRV62TSYMM7TANCNFSM4VCINHVA>.
>
I can't tell you for sure: Over the years I added the following to my
entry in /etc/user_attr:
...,dtrace_user,file_dac_read,file_dac_write,proc_clock_highres;profiles=Primary
Administrator;roles=root

But still the tests run flaky. The newly created interfaces won't be
removed at the end of a gmake test and create warnings when I run it again.

**Reply by balrog-nona** [reply]
Yes, the problem was with my user settings. I added Primary Administrator to my profile. Thank you!

**Comment by balrog-nona** [comment]
Hi Andreas,
I ran `gmake test` and on my machine, there is one more failed test:
```
Test results in /export/home/nona/oi-userland/components/network/bind/build/test/amd64/test-64-results
Differences found.
--- /export/home/nona/oi-userland/components/network/bind/test/results-64.master	2021-01-05 20:51:15.978954878 +0000
+++ /export/home/nona/oi-userland/components/network/bind/build/test/amd64/results-64.snapshot	2021-01-06 14:02:16.631859916 +0000
@@ -1,9 +1,10 @@
 R:cds:FAIL
+R:legacy:FAIL
 R:nsupdate:FAIL
 R:tsig:FAIL
 R:tsiggss:FAIL
 I:System test result summary:
-I:      4 FAIL
+I:      5 FAIL
 I:     74 PASS
 I:      3 SKIPPED
-I:     19 UNTESTED
+I:     18 UNTESTED
gmake: *** [/export/home/nona/oi-userland/make-rules/configure.mk:222: /export/home/nona/oi-userland/components/network/bind/build/amd64/.tested-and-compared] Error 2
```
When running `gmake publish`, I get this warning:
```Lint engine setup...
Starting lint run...
INFO pkglint001.5                 Linted message: pkglint.manifest004  last name component bind in package name clashes across pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0 pkg:/service/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/database/lmdb@0.9.23-2020.0.1.0 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/library/json-c@0.12-2020.0.1.0 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/library/libedit@3.1.20181209-2020.0.1.2 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/library/libuv@1.34.2-2020.0.1.0 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/library/libxml2@2.9.9-2020.0.1.4 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/library/security/openssl@1.0.2.21-2020.0.1.2 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/library/zlib@1.2.11-2020.0.1.3 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/service/security/kerberos-5@0.5.11-2020.0.1.20221 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/system/library/security/gss@0.5.11-2020.0.1.20221 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/system/library@0.5.11-2020.0.1.20221 for pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0
INFO pkglint001.5                 Linted message: pkglint.manifest004  last name component bind in package name clashes across pkg:/network/dns/bind@9.16.10,5.11-2020.0.1.0 pkg:/service/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/SUNWcs@0.5.11-2020.0.1.20221 for pkg:/service/network/dns/bind@9.16.10,5.11-2020.0.1.0
WARNING pkglint.action005.1       obsolete dependency check skipped: unable to find dependency pkg:/runtime/python-35@3.5.10-2020.0.1.0 for pkg:/service/network/dns/bind@9.16.10,5.11-2020.0.1.0
```
Yet, I'm able to publish it and install it locally. 

I'm afraid, I'm able to test it just using bind's dig command. 
When I run `dig oracle.com`, I get this:
```
; <<>> DiG 9.16.10 <<>> oracle.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 12850
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;oracle.com.                    IN      A

;; ANSWER SECTION:
oracle.com.             300     IN      A       137.254.120.50

;; Query time: 0 msec
;; SERVER: 108.61.10.10#53(108.61.10.10)
;; WHEN: Wed Jan 06 14:51:45 CET 2021
;; MSG SIZE  rcvd: 55
```
which seems correct to me. 

Trying `dig oracle.com +short`, I get this:
```
137.254.120.50
```
so it also works.

Could you please tell me, what exact further checks I should perform with it to provide more relevant information?

As usual, the dig's man page says the right version :)
